lib/dot-compat.js

```diff
@@ -0,0 +1,298 @@
+    const keysToCopy = ["settings", "rules", "processor"];
+    const languageOptionsKeysToCopy = ["globals", "parser", "parserOptions"];
+    const linterOptionsKeysToCopy = ["noInlineConfig", "reportUnusedDisableDirectives"];
+    const pluginEnvironments = new Map();
+
+    // check for special settings for eslint:all and eslint:recommended:
+    if (eslintrcConfig.settings) {
+        if (eslintrcConfig.settings["eslint:all"] === true) {
+            return ["eslint:all"];
+        }
+
+        if (eslintrcConfig.settings["eslint:recommended"] === true) {
+            return ["eslint:recommended"];
+        }
+    }
+
+    // copy over simple translations
+    for (const key of keysToCopy) {
+        if (key in eslintrcConfig && typeof eslintrcConfig[key] !== "undefined") {
+            dotConfig[key] = eslintrcConfig[key];
+        }
+    }
+
+    // copy over languageOptions
+    for (const key of languageOptionsKeysToCopy) {
+        if (key in eslintrcConfig && typeof eslintrcConfig[key] !== "undefined") {
+            dotConfig.languageOptions = languageOptions;
+
+            if (languageOptions[key] && typeof languageOptions[key] === "object") {
+                languageOptions[key] = {
+                    ...eslintrcConfig[key]
+                };
+            } else {
+                languageOptions[key] = eslintrcConfig[key];
+
+                if (key === "parser") {
+                    debug(`Resolving parser '${languageOptions[key]}' relative to ${resolveConfigRelativeTo}`);
+                    languageOptions[key] = eslintrcConfig[key].definition;
+                }
+            }
+        }
+    }
+
+    // copy over linterOptions
+    for (const key of linterOptionsKeysToCopy) {
+        if (key in eslintrcConfig && typeof eslintrcConfig[key] !== "undefined") {
+            dotConfig.linterOptions = linterOptions;
+            linterOptions[key] = eslintrcConfig[key];
+        }
+    }
+
+    // move ecmaVersion a level up
+    if (languageOptions.parserOptions) {
+
+        if ("ecmaVersion" in languageOptions.parserOptions) {
+            languageOptions.ecmaVersion = languageOptions.parserOptions.ecmaVersion;
+            delete languageOptions.parserOptions.ecmaVersion;
+        }
+
+        if ("sourceType" in languageOptions.parserOptions) {
+            languageOptions.sourceType = languageOptions.parserOptions.sourceType;
+            delete languageOptions.parserOptions.sourceType;
+        }
+
+        // check to see if we even need parserOptions anymore and remove it if not
+        if (Object.keys(languageOptions.parserOptions).length === 0) {
+            delete languageOptions.parserOptions;
+        }
+    }
+
+    if (eslintrcConfig.ignorePattern) {
+        configs.unshift({
+            ignores: eslintrcConfig.ignorePattern.patterns
+        });
+    }
+
+    // overrides
+    if (eslintrcConfig.criteria) {
+
+        for (const { includes, excludes } of eslintrcConfig.criteria.patterns) {
```

Comment:
This loop looks a wrong conversion. If the `patterns` values are multiple, it must be AND operation. For example, if `[{ includes: ["src/**"] }, { includes: ["*.ts"] }]` is present, then it matches `src/foo.ts`, but not `src/foo.js` and `dist/foo.ts`.

Reply:
Also, note the `criteria.patterns` values are relatived from `criteria.basePath`.

Reply:
Okay, I'll take a look at that. Thanks.